.circleci/config.yml

```diff
@@ -0,0 +1,50 @@
+---
+version: 2
+
+jobs:
+  build:
+    working_directory: /tmp/code
+    docker:
+      - image: docker:18.02.0-ce
+    environment:
+      TZ: "/usr/share/zoneinfo/Etc/UTC"
+      TERM: dumb
+      DEBIAN_FRONTEND: "noninteractive"
+      # Definition of variables using other variables is not supported yet
+      # on CircleCi 2.0
+      # https://discuss.circleci.com/t/circle-2-0-global-environment-variables/8681/11
+      # For that reason we are storing the definition of variables
+      # in `variables` file - see `.circleci/variables`
+
+    steps:
+      # Checkout code
+      - checkout
+      # https://circleci.com/docs/2.0/building-docker-images/
+      - setup_remote_docker
+      - run:
+          name: Create a volume
+          command: |
+            docker create -v /src -v /mnt --name src alpine:3.7 /bin/true
+            docker cp . src:/src
+      - run:
+          name: Linting
+          command: |
+            docker run --rm -i --volumes-from src singapore/lint-condo /bin/sh -c \
+              "find . -name '*.md' -print0 | xargs -r -0 proselint || true"
+            docker run --rm -i --volumes-from src singapore/lint-condo /bin/sh -c \
+              "find . -name '*.md' -print0 | xargs --no-run-if-empty -0 remark"
+            docker run --rm -i --volumes-from src singapore/lint-condo /bin/sh -c \
+              "find . -name '*.md' -print0 | xargs --no-run-if-empty -0 write-good || true"
+            docker run --rm -i --volumes-from src singapore/lint-condo /bin/sh -c \
+              "find . -name '*.yml' -print0 | xargs --no-run-if-empty -0 yamllint"
+            docker run --rm -i --volumes-from src singapore/lint-condo /bin/sh -c \
+              "find . -name '*.yaml' -print0 | xargs --no-run-if-empty -0 yamllint"
+            for i in kube-tail kube-cluster kube-dash kube-ns kubectl-plugins/tail/tail.sh
```

Comment:
Generally I prefer having most tasks runnable both locally and in circleCI using the same mechanism so that engineers can run tests locally as well as in the CI pipeline. This gets a bit more difficult with docker-based circle2.0 workflows though because of the remote_docker and lack of volume mounting. I intend to bring your changes in but I want to find a way to do it such that `make [lint,test]` can be used both locally and in circle. The volumes-from + cp trick seems like the way forward and it's just a matter of implementing that in the makefile now.

In some cases we end up with two targets like `make [test, test-circle]` where the test-circle target wraps any circle-specific setup before calling the test target. I hope to avoid that but it's another option.